ckanext/dia_theme/templates/package/search.html

```diff
@@ -76,6 +76,13 @@ <h2>Error</h2>
 {% endblock %}
 
 
+{# Add a link to add datasets below search results #}
+{% block package_search_results_list %}
+  {{ super() }}
+  <div class="request-data">Can't find what you're looking for? <a href="{{ h.parent_site_url() }}/request-data/" title="Request a new dataset through data.govt.nz">Request a dataset.</a></div>
```

Comment:
may need to wrap this in a notification box - suggest the "yellow" colour/warning (not red or green, blue will blend too much).